### PR TITLE
fix: remove conformance of UIControl.State to Hashable [WPB-6969]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6969" title="WPB-6969" target="_blank">BugWPB-6969</a>  One on One call and video fails
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

By using its raw value there is no reason that `UIControl.State` conforms to `Hashable`.
This PR removes the extension.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/Components/Buttons/LegacyButton.swift [hunk 1]
```diff
@@ -86,6 +86,7 @@ class LegacyButton: ButtonWithLargerHitArea {
     var textTransform: TextTransform = .none {
         didSet {
             for(state, title) in originalTitles {
+                let state = UIControl.State(rawValue: state)
                 setTitle(title, for: state)
             }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/Components/Buttons/LegacyButton.swift]:
```suggestion
            for(stateRawValue, title) in originalTitles {
                let state = UIControl.State(rawValue: stateRawValue)
                setTitle(title, for: state)
            }
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Helpers/IconButton+Factory.swift [hunk 2]
```diff
@@ -94,9 +96,3 @@ extension IconButton {
     }
 
 }
-
-extension UIControl.State: Hashable {
-    public func hash(into hasher: inout Hasher) {
-        hasher.combine(self.rawValue)
-    }
-}
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Helpers/IconButton+Factory.swift:102]: Thank god! 💪

File: wire-ios/Wire-iOS/Sources/UserInterface/Helpers/IconButton+Factory.swift [hunk 3]
```diff
@@ -78,10 +78,12 @@ extension IconButton {
...
 
         for (state, color) in backgroundColor {
+            let state = UIControl.State(rawValue: state)
             setBackgroundImageColor(color, for: state)
         }
 
         for (state, color) in iconColor {
+            let state = UIControl.State(rawValue: state)
             setIconColor(color, for: state)
         }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Helpers/IconButton+Factory.swift]:
```suggestion
        for (state, color) in iconColor {
            setIconColor(color, for: .init(rawValue: state))
        }
```

Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Helpers/IconButton+Factory.swift]: Or rather like this?

**Review by markusfassbender — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Pull request merged**